src/cascade/model/model_reader.py

```diff
@@ -0,0 +1,313 @@
+from cascade.model.var import Var
+
+CODELOG, MATHLOG = getLoggers(__name__)
+
+
+def write_vars(dismod_file, new_vars, var_ids, which):
+    """
+
+    Args:
+        dismod_file:
+        new_vars (DismodGroups): The new vars to write.
+        var_ids (DismodGroups): The output of ``read_var_table_as_id``.
+        which (str): Could be "start_var", "truth_var", "scale_var", "fit_var".
+    """
+    var_name = f"{which}_var"
+    id_column = f"{var_name}_id"
+    total = list()
+    for group_name, group in var_ids.items():
+        for key, value in group.items():
+            total.append(_construct_vars_one_field(var_name, value, new_vars[group_name][key]))
+    new_table = pd.concat(total).sort_values(by=[id_column])
+    setattr(dismod_file, var_name, new_table)
+
+
+def _construct_vars_one_field(name, var_id, new_var):
+    id_column = f"{name}_id"
+    var_column = f"{name}_value"
+    with_id = new_var.grid.merge(var_id.grid[["age", "time", "var_id"]], on=["age", "time"], how="left")
+    if with_id.var_id.isna().any():
+        raise RuntimeError(f"Internal error: Could not align {id_column} with var_id.")
+    return pd.DataFrame({
+        id_column: with_id.var_id.values,
+        var_column: with_id["mean"].values,
+    })
+
+
+def read_vars(dismod_file, var_ids, which):
+    """
+    Reads a table full of vars (start, truth, scale).
+
+    Args:
+        dismod_file:
+        var_ids (DismodGroups): The output of ``read_var_table_as_id``.
+        which (str): Could be "start_var", "truth_var", "scale_var", "fit_var".
+
+    Returns:
+        DismodGroups: With means for everything.
+    """
+    var_name = f"{which}_var"
+    table = getattr(dismod_file, var_name)
+    return _assign_from_var_ids(table, var_ids, partial(_read_vars_one_field, name=var_name))
+
+
+def _assign_from_var_ids(table, var_ids, assignment):
+    """Iterate over var ids and execute a function on each grid in the varids."""
+    if table.empty:
+        raise AttributeError(f"Dismod file has no data in table {table.columns}.")
+    var_groups = DismodGroups()
+    for group_name, group in var_ids.items():
+        for key, value in group.items():
+            var_groups[group_name][key] = assignment(table, value)
+    return var_groups
+
+
+def _read_vars_one_field(table, id_draw, name):
+    """Read one rate or randome effect's values.
+
+    Args:
+        table: Dismod-AT table represented by a Pandas dataframe.
+        id_draw (AgeTimeGrid): An AgeTimeGrid that contains rows of ["var_id"].
+        name (str): name of the table, following Dismod-AT conventions.
+
+    Returns:
+        A new Var containing the values at these ages and times.
+    """
+    id_column = f"{name}_id"
+    var_column = f"{name}_value"
+    table = table.reset_index(drop=True)
+    with_var = id_draw.grid.merge(table, left_on="var_id", right_on=id_column, how="left")
+
+    vals = Var((id_draw.ages, id_draw.times))
+    vals.grid = vals.grid.drop(columns=["mean"]) \
```

Comment:
I think this is a fine way to do this but I believe it would be clearer to me as index alignment followed by assignment rather than a merge surrounded by drop and rename

Reply:
Ahah! I can do this with a straight-up assign statement!